systemservices/marketplace/src/contracts/version.ts

```diff
@@ -10,33 +10,70 @@ const getServiceVersions = async (contract: Marketplace, sid: string): Promise<V
     return []
   }
   const versionsLength = new BigNumber(await contract.methods.serviceVersionsLength(sidHex).call())
-  const versionsPromise: Promise<Version|undefined>[] = []
+  const versions: Version[] = []
   for (let j = new BigNumber(0); versionsLength.isGreaterThan(j); j = j.plus(1)) {
-    versionsPromise.push(getServiceVersionWithIndex(contract, sid, j))
+    try {
+      const version = normalizeVersion(await getServiceVersionWithIndex(contract, sid, j))
```

Comment:
please fetch the version in a parallel way

Reply:
Parallel here makes it really complicated to read with some mapping and filtering.
That's the problem of async either we use full async/await or full promise but mixing is really hard to read. But I can put the parallel version and we can improve later